src/sys/unix/dlsym.rs

```diff
@@ -10,7 +10,7 @@ macro_rules! dlsym {
         static $name: ::sys::unix::dlsym::DlSym<unsafe extern fn($($t),*) -> $ret> =
             ::sys::unix::dlsym::DlSym {
                 name: concat!(stringify!($name), "\0"),
-                addr: ::std::sync::atomic::ATOMIC_USIZE_INIT,
+                addr: ::std::sync::atomic::AtomicUsize::new(0),
```

Comment:
Remove this, mio should still work with older versions of Rust. (I'm aware this creates a deprecation warning).

Reply:
@Thomasdezeeuw I made this change on master. Mio 0.7 can target a modern Rust :+1: